gflownet/envs/crystals/composition.py

```diff
@@ -258,6 +287,67 @@ def get_element_mask(min_atoms, max_atoms):
             min_atoms_per_unrequired_element, max_atoms_per_unrequired_element
         )
 
+        # If required, refine the masks by doing compatibility checks between
+        # the space group and the number of atoms
+        if self.do_spacegroup_check and isinstance(self.space_group, int):
+            space_group = get_space_group(self.space_group)
+            n_atoms = [s for s in state if s > 0]
+
+            # Get the greated common divisor of the group's wyckoff position.
+            # It cannot be valid to add a number of atoms that is not a
+            # multiple of this value
+            wyckoff_gcd = space_group_wyckoff_gcd(self.space_group)
+
+            # Get the multiplicity of the group's most specific wyckoff position with
+            # at least one degree of freedom
+            free_multiplicity = space_group_lowest_free_wp_multiplicity(
+                self.space_group
+            )
+
+            # Go through each action in the masks, validating them
+            # individually
+            for action_idx, nb_atoms_action in enumerate(
+                range(self.min_atom_i, self.max_atom_i + 1)
```

Comment:
Is it correct that here we should use `self.min_atom_i` and `self.max_atom_i` instead of `min_atom_i` and `max_atom_i` that are updated above?

Reply:
No because, here, the goal is to go over all the actions in the masks (`mask_required_element `and `mask_unrequired_element`). And, currently, these masks have the same length whatever the space group.
We could change the implementation where the masks are only as long as necessary given the space group but that would involve changing the `get_element_mask()` function above, and the `for` loop below to copy those element masks into the global mask afterwards.

Reply:
I see, that makes sense. Let's leave it as is then. Then my changes [here](https://github.com/alexhernandezgarcia/gflownet/commit/48df853d64f8078430df2e05d17955c798772550) should be fine enough, I hope.